src/runner/bootstrapper.ts

```diff
@@ -233,14 +217,25 @@ export default class Bootstrapper {
         return tests.filter(test => predicate(test.name, test.fixture.name, test.fixture.path, test.meta, test.fixture.meta));
     }
 
+    private async _compileTests ({ sourceList, compilerOptions }: CompilerArguments): Promise<Test[]> {
+        if (!this.compilerService) {
+            const compiler = new Compiler(sourceList, compilerOptions);
```

Comment:
We can't do it right now. Unlike `Compiler`, `CompilerHost` constructor is called only once during the TestCafe lifetime, so we have to pass the list of test files to `getTests` instead of the constructor. And I would like to avoid modifying `Compiler`'s interface until this feature comes out of the experimental state to minimize the risk of breaking changes.

I will remove the `CompilerProvider` interface since I failed to properly implement and use it due to the reasons described above.

Reply:
I don't mind leaving it as it is for now at least, if you don't want to modify `Compiler`'s interface.